### DISCUSSION TITLE
Trying to build with PMTiles support

### DISCUSSION THREAD

**Discussion by scriptprojectsdev** [maplibre/maplibre-native#3025] [Q&A] [answered]
@tdcosta100 I'm trying to build the library with pmtiles support. I merged in your PR to the main branch and followed the build steps here: https://github.com/maplibre/maplibre-native/blob/main/platform/ios/CONTRIBUTING.md 
I modified the App target to use a style that uses a `pmtiles://` source and it returns an error saying `unsupported URL`

The full style file I'm using is below. I'm doing something wrong? Is the port in the URL the issue? I checked that the `MLN_WITH_PMTILES` CMake option is on. Do I need to do anything else to enable support?

```json
{
  "version": 8,
  "name": "Harris County Parcels",
  "sources": {
    "parcels-source": {
      "type": "vector",
      "url": "pmtiles://192.168.50.231:8080/harris.pmtiles"
    }
  },
  "layers": {
    "id": "parcels",
    "type": "line",
    "source": "parcels-source",
    "source-layer": "stratmap24-landparcels_48201_harris_202407",
    "layout": {},
    "paint": {
      "line-color": "#aaa"
    }
  }
}
```

**Comment by tdcosta100** [comment]
Hi, @scriptprojectsdev. Before trying to figure out if there's something wrong with the build, please try to correct your URL using this: `pmtiles://http://192.168.50.231:8080/harris.pmtiles`. The PMTiles provider just remove the `pmtiles://` part of the URL and then forwards it to the next provider available. Since you are not putting anything after `pmtiles://` as protocol, none provider accepts it.

**Reply by scriptprojectsdev** [reply]
That did get rid of the error. The data is not rendering though. Will debug further tomorrow.

**Reply by tdcosta100** [reply]
Nice! I hope you will be able to use PMTiles as well as MBTiles.

**Reply by scriptprojectsdev** [reply]
I wasn't able to get it to render and I even tried using the protomaps API which shouldn't require pmtiles support (AFAIK). The styleURL is in the format `https://api.protomaps.com/styles/v2/light.json?key={API_KEY}` . Maybe I'm doing something wrong.

**Reply by tdcosta100** [reply]
Yeah, maybe. You should try with the simplest examples, like the ones which don't require an API key and fetch the tiles directly by URL.

**Comment by acalcutt** [comment]
I created this protomaps test style that seems like it works for me locally, if you wanted to try it and see what you get with it.
[protomaps_style.json](https://github.com/user-attachments/files/17821100/protomaps_style.json)

I used this command
` xvfb-run --auto-servernum ./build/bin/mbgl-render  --style /opt/maplibre-native/protomaps_style.json --zoom=7 --lat 42.378033 --lon -71.9265767`
and it made this 'out.png' image
![out](https://github.com/user-attachments/assets/590954b3-344c-46b5-991e-7dca64227be4)

I built on ubuntu 22.04 using the instructions from 
https://github.com/maplibre/maplibre-native/tree/main/platform/linux 

I checked the pmtiles test branch using the following
```
git clone --recurse-submodules -j8 https://github.com/maplibre/maplibre-native.git
cd maplibre-native
git remote add tdcosta100 https://github.com/tdcosta100/maplibre-native.git
git fetch tdcosta100
git checkout tdcosta100/pmtiles-support
git submodule sync
git submodule update
```

**Reply by scriptprojectsdev** [reply]
I tried the style URL and it didn't work in the app. I might try these commands to see if I'm able to get the image you got as well.

**Reply by acalcutt** [reply]
I tested using the url above and that also worked for me
`xvfb-run --auto-servernum ./build/bin/mbgl-render  --style https://github.com/user-attachments/files/17821100/protomaps_style.json --zoom 6 --lat 42.378033 --lon -71.9265767`

**Reply by acalcutt** [reply]
When you say app, are you trying to use this with ios or android builds of maplibre-native or something else?

**Reply by acalcutt** [reply]
It looks like you are using ios from your first post, I think to build that version it uses bazel instead of cmake.. I haven't tried that myself, 

Looking at the PR I don't see anything like like the MLN_WITH_PMTILES cmake flag in the bazel configs....or really anything for ios @tdcosta100  am I missing it or is there no ios support right now?

**Reply by tdcosta100** [reply]
Hi @acalcutt. AFAIK, PMTiles is included in iOS builds, but this needs to be checked by someone with iOS build framework.

**Reply by scriptprojectsdev** [reply]
I'm using the demo app that's built using bazel. I was getting an error when I didn't use the right url, once I did that there was no error, so I'm guessing the pmtiles support is there, but I'm not sure how to test that. It doesn't seem to be working.

**Reply by tdcosta100** [reply]
@scriptprojectsdev, could you please test with this URL?

https://raw.githubusercontent.com/maplibre/demotiles/gh-pages/style.json

It doesn't use PMTiles but it will be useful to see if it renders something at least.

**Reply by scriptprojectsdev** [reply]
Yeah, @tdcosta100 that one works. That's the style that was initially there and I replaced with the pmtiles one.

**Reply by scriptprojectsdev** [reply, marked as answer]
As of January 2025 map-libre supports pmtiles natively 🙌
